Source/CronosPlayUnreal/Private/PlayCppSdkActor.cpp

```diff
@@ -325,13 +469,19 @@ void APlayCppSdkActor::SignPersonal(FString user_message, TArray<uint8> address,
 }
 
 void APlayCppSdkActor::SignEip155Transaction(
-    FWalletConnectTxEip155 info, TArray<uint8> address,
+    FWalletConnectTxEip155 info,
     FWalletconnectSignEip155TransactionDelegate Out) {
-
-  ::com::crypto::game_sdk::WalletconnectClient *coreclient = _coreClient;
-  assert(coreclient != NULL);
+  ::com::crypto::game_sdk::WalletconnectClient *coreclient = GetClient();
+  // if no walletconnect session, return
+  if (coreclient == nullptr)
+    return;
+  TArray<uint8> address = GetAddress();
+  int64 chain_id = (uint64)GetChainId();
+  // if no address, return
+  if (address.Num() == 0 || chain_id == 0)
```

Comment:
I think chainId == 0 is that the wallet will choose it ? or is it expected chain id should have been returned and saved during the init?

Reply:
if chain_id is invalid, return false?

Reply:
chain_id is got from EnsureSession which is the return value from WalletConnect after successfully connected. So it is impossible to be zero, if it is successfully connected. If it is 0, it means, the connection is not built up.